.github/workflows/github-actions.yml

```diff
@@ -0,0 +1,158 @@
+name: Test and Publish Sonar Results
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+
+jobs:
+ build_visualization_test:
+    name: Visualization Test
+    runs-on: ubuntu-latest
+    env:
+      working-directory: ./visualization
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Setup Node.js environment
+      uses: actions/setup-node@v2.1.4
+      with:
+        node-version: '12'  
+
+    - name: Cache node modules
+      uses: actions/cache@v2
+      env:
+        cache-name: cache-node-modules
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
+    - name: Install Node
+      run: npm install
+      working-directory: ${{env.working-directory}} 
+
+    - name: Recache-Node     
+      if: failure()
+      run: |
+        npm cache clean --force
+        npm cache verify
+      working-directory: ${{env.working-directory}}
```

Comment:
This seems rather surprising to me? Do we need this part?

Reply:
I will combine a bit more, thats a good point
Concerning the marked line, it was a work around for the nw-builder failure as simply caching it would result in the build failing again when it was restarted manually. We can remove this line now

Reply:
Concerning redundancy, we felt that it would make the jobs more modular as every job would handle his own problem, independent of others.
But maybe that is not the correct philosophy for this task.

Reply:
If I understood correct, the whole re-cache part may be removed?